File: Longest Subsequence with Adjacent Diff as 1.cpp
// Problem: Longest Subsequence with Adjacent Diff as 1
// geeksforgeeks problem of the day -> 5th September 2026
// C++ CODE
class Solution {
public:
    int longestSubseq(vector<int>& arr) {
        unordered_map<int, int> dp;
        int ans = 0;
        for (int x : arr) {
            int prev = max(dp[x - 1], dp[x + 1]);
            dp[x] = prev + 1;
            ans = max(ans, dp[x]);
        }
        return ans;
    }
};

File: Longest Subsequence with Adjacent Diff as 1.java
// Problem: Longest Subsequence with Adjacent Diff as 1
// geeksforgeeks problem of the day -> 5th September 2026
// JAVA CODE
class Solution {
    public int longestSubseq(int[] arr) {
        Map<Integer, Integer> map = new HashMap<>();
        int ans = 0;
        
        for(int x: arr){
            int best = Math.max(
                map.getOrDefault(x-1, 0),
                map.getOrDefault(x+1, 0)
                );
            map.put(x, best+1);
            
            ans = Math.max(ans, map.get(x));
        }
        return ans;
    }
}